### PR TITLE
Add QE overlay for control plane deployment

### PR DESCRIPTION
**Pull request by asmigala**
## Description
Add kustomize overlay for deploying control plane on QE clusters.

Adds necessary changes to SSO for admin API to function, including a k8s job to create the SSO users.

## Verification Steps
Follow instructions in `etc/kubernetes/manifests/overlays/qe/README.md`

## Checklist (Definition of Done)

- [ ] ~~All acceptance criteria specified in JIRA have been completed~~
- [ ] ~~Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)~~
- [x] Documentation added for the feature
- [ ] ~~CI and all relevant tests are passing~~
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~
- [ ] ~~JIRA has created for changes required on the client side~~

### CODE REVIEW

File: etc/kubernetes/manifests/overlays/local/keycloak/route.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: route.openshift.io/v1
+kind: Route
```
Comment (lburgazzoli) [inline etc/kubernetes/manifests/overlays/local/keycloak/route.yaml:2]: the local overlay is meant to be deployed on plain kubernetes, so I guess this should be part of the qe overlay

File: etc/kubernetes/manifests/overlays/local/keycloak/data/create-user.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -e
+set -x
+
+# wait for keycloak to deploy
+kubectl wait --for=condition=Available deployment/cos-fleet-manager-kc
+KC_BASE_PATH=https://$(kubectl get route cos-fleet-manager-kc -o jsonpath={.spec.host})
```
Comment (lburgazzoli) [inline etc/kubernetes/manifests/overlays/local/keycloak/data/create-user.sh:8]: as this script is executed inside the cluster, do we need the route ? I guess the service would be enough.
eventually the path of the service  can be provided as an argument of the script

**Review by valdar — APPROVED** [review]

File: etc/kubernetes/manifests/overlays/local/keycloak/data/create-user.sh (hunk 2 above)
Comment (asmigala) [reply etc/kubernetes/manifests/overlays/local/keycloak/data/create-user.sh:8]: We are also setting the frontend URL of the realms to the route on lines 34-37, to allow login using the rhoc tool

File: etc/kubernetes/manifests/overlays/local/keycloak/deployment.yaml [hunk 3]
```diff
@@ -26,8 +26,8 @@ spec:
               value: "admin"
             - name: JAVA_OPTS_APPEND
               value: "-Dkeycloak.import=/opt/keycloak/data/import/rhoas-realm.json,/opt/keycloak/data/import/rhoas-kafka-sre-realm.json"
-            - name: KC_HOSTNAME
-              value: "cos-fleet-manager-kc"
+            - name: KC_PROXY
+              value: "edge"
```
Comment (asmigala) [inline etc/kubernetes/manifests/overlays/local/keycloak/deployment.yaml:30]: @lburgazzoli if I remove the route, I'll probably also have to revert this. Makes me wonder whether I should just move all the changes to the qe overlay, including the user script and job, and just patch the keycloak deployment there.

Comment (lburgazzoli) [reply etc/kubernetes/manifests/overlays/local/keycloak/deployment.yaml:30]: I think for the moment  yes, that is probably the best option. 
We should probably explore if we can do the same locally by using an ingress

**Review by lburgazzoli — APPROVED** [review]

**Pull request merged**